bigbluebutton-html5/imports/ui/services/LocalCollectionSynchronizer/LocalCollectionSynchronizer.js

```diff
@@ -31,26 +32,12 @@ class LocalCollectionSynchronizer {
     const self = this;
 
     const addedCallback = function (item) {
-      const subscription = SubscriptionRegistry
-        .getSubscription(self.serverCollection._name);
       if (item.id === 'publication-stop-marker' && item.stopped) {
         self.ignoreDeletes = true;
         return;
       }
-      // If the subscriptionId changes means the subscriptions was redone
-      // or theres more than one subscription per collection
-      if (subscription && (self.lastSubscriptionId !== subscription.subscriptionId)) {
-        const wasEmpty = self.lastSubscriptionId === '';
-        self.lastSubscriptionId = subscription.subscriptionId;
-        if (!wasEmpty) {
-          self.callWhenSubscriptionReady(() => {
-            self.ignoreDeletes = false;
-            Session.set('globalIgnoreDeletes', false);
-            self.removeOldSubscriptionData();
-          });
-        }
-      }
 
+      self.checkForStaleData();
```

Comment:
Could you add a throttle for this function? I think thats good have it, but I think it could add performance issues call it while the client is sync bacause it would generates a lot of calls

Reply:
> Could you add a throttle for this function? I think thats good have it, but I think it could add performance issues call it while the client is sync bacause it would generates a lot of calls

I could add a tail debounce, yes. Although I'd like to clear some things up:
  - I don't think it'll "add" performance issues because it's virtually the same as the status quo (v2.5.x/v2.6.x have the same code block as this hooked to the `added` callback - the only difference here is that I refactored it to a method to be re-usable)
  - From the PR: `The added check was also maintained, although I'm not too sure it's still needed. That may need to be revisited`. I'm under the impression this doesn't need to be in the `added` callback anymore if it is run once the subscription is set up. If you can confirm that is the case, I'll just remove it.